dask_match/expr.py

```diff
@@ -169,6 +169,22 @@ def dependencies(self):
         # Dependencies are `Expr` operands only
         return [operand for operand in self.operands if isinstance(operand, Expr)]
 
+    @property
+    def _culled(self) -> bool:
```

Comment:
A thought, rather than add this to all Exprs, maybe we have a `PartitionsFiltered` mixin class that looks like the following:

```python
class PartitionsFiltered(Expr):
    def _partitions(self):
        ...
    def _divisions(self):
        ...
    def _layer(self):
        ...
```

And whatever other methods need modification.  Then we add this mixin to the IO classes

```python
class BlockwiseIO(Blockwise, PartitionsFiltered):
    ...
```

This is probably a bit more code, but my guess is that it isolates the oddity of `_partitions` to just those classes that are affected by it.